packages/dev-server-core/src/server/DevServer.ts

```diff
@@ -39,6 +40,8 @@ export class DevServer {
           this.connections.delete(connection);
         });
       });
+    } else if (externalServer) {
+      this.webSockets = new WebSocketsManager(externalServer);
```

Comment:
is `WebSocketsManager` compatible with the Storybook manager server? with other servers?
I'm not aware of how WebSockets layer works on Storybook or WDS side

Reply:
Storybook is using "connect" / "express" API
WDS used koa API
they do have compatibility issues

for middleware I was able to find a way to make it work without any special compatibility layer
maybe it's simple here too

Reply:
The only responsibility of the `server` in `WebSocketsManager` appears to be to have an `on()` method that can listen for an `upgrade` event. This is on the shared type, so, hopefully, any compatibility issues would have to be quite serious to effect this cross-over?

Reply:
I don't get though how this reloads the page?

Reply:
This watches any files that the server has served: https://github.com/modernweb-dev/web/blob/master/packages/dev-server-core/src/middleware/watchServedFilesMiddleware.ts